libdnf/rpm/transaction.cpp

```diff
@@ -0,0 +1,746 @@
+    /// @return  transaction root directory
+    const char * get_root_dir() const { return rpmtsRootDir(ts); }
+
+    /// Retrieve color bits of transaction set.
+    /// @return  color bits
+    rpm_color_t get_color() const { return rpmtsColor(ts); }
+
+    /// Set color bits of transaction set.
+    /// @param color  new color bits
+    /// @return  previous color bits
+    rpm_color_t set_color(rpm_color_t color) { return rpmtsSetColor(ts, color); }
+
+    /// Retrieve preferred file color
+    /// @return  color bits
+    rpm_color_t get_pref_color() const { return rpmtsPrefColor(ts); }
+
+    /// Set preferred file color
+    /// @param color  new color bits
+    /// @return  previous color bits
+    rpm_color_t set_pref_color(rpm_color_t color) { return rpmtsSetPrefColor(ts, color); }
+
+    /// Get transaction flags, i.e. bits that control rpmtsRun().
+    /// @return  transaction flags
+    rpmtransFlags get_flags() const { return rpmtsFlags(ts); }
+
+    /// Set transaction flags, i.e. bits that control rpmtsRun().
+    /// @param flags  new transaction flags
+    /// @return  previous transaction flags
+    rpmtransFlags set_flags(rpmtransFlags flags) { return rpmtsSetFlags(ts, flags); }
+
+    /// Get verify signatures flag(s).
+    /// @return  verify signatures flags
+    rpmVSFlags get_signature_verify_flags() const { return rpmtsVSFlags(ts); }
+
+    /// Set verify signatures flag(s).
+    /// @param verify_flags  new verify signatures flags
+    /// @return         previous value
+    rpmVSFlags set_signature_verify_flags(rpmVSFlags verify_flags) { return rpmtsSetVSFlags(ts, verify_flags); }
+
+    /// Get package verify flag(s).
+    /// @return  verify signatures flags
+    rpmVSFlags get_pkg_verify_flags() const { return rpmtsVfyFlags(ts); }
+
+    /// Set package verify flag(s).
+    /// @param verify_flags  new package verify flags
+    /// @return  old package verify flags
+    rpmVSFlags set_pkg_verify_flags(rpmVSFlags verify_flags) { return rpmtsSetVfyFlags(ts, verify_flags); }
+
+    /// Get enforced package verify level
+    /// @return  package verify level
+    int get_pkg_verify_level() const { return rpmtsVfyLevel(ts); }
+
+    /// Set enforced package verify level
+    /// @param verify_level  new package verify level
+    /// @return  old package verify level
+    int set_pkg_verify_level(int verify_level) { return rpmtsSetVfyLevel(ts, verify_level); }
+
+    /// Get transaction id, i.e. transaction time stamp.
+    /// @return  transaction id
+    rpm_tid_t get_id() const { return rpmtsGetTid(ts); }
+
+    /// Return header from package.
+    /// @param path  file path
+    /// @retval hdrp  header (NULL on failure)
+    /// @return  package header
+    Header read_pkg_header(const std::string & file_path) const {
+        FD_t fd = Fopen(file_path.c_str(), "r.ufdio");
+
+        if (!fd) {
+            throw Exception("read_pkg_header: Can't open file: " + file_path);
+        }
+
+        Header h;
```

Comment:
clang-tidy complains that it should be initialized to nullptr

Reply:
`h` is initialized two lines later:
`rpmRC rpmrc = rpmReadPackageFile(ts, fd, descr, &h);`
`h` is output argument of the function. But I understand that clang-tidy does not know this information.
I will change the code to satisfy the `clang-tidy`. And I will add comment that it is not needed.